src/features/fingerprinting-hardware.js

```diff
@@ -1,28 +1,19 @@
-import { overrideProperty } from '../utils'
 import ContentFeature from '../content-feature'
+import { wrapProperty } from '../wrapper-utils'
 
 export default class FingerprintingHardware extends ContentFeature {
     init () {
-        const Navigator = globalThis.Navigator
-        const navigator = globalThis.navigator
-
-        overrideProperty('keyboard', {
```

Comment:
I don't really understand the move from `overrideProperty` to wrap other than it's arguments are different and the API is different (this makes the diff much more noisy than it probably needs to be).

Perhaps there's more justification in the work you're doing, but as it stands I would keep the same function signature for all this code.

Can we rename it back and keep the same args, what's the real change this `toString` that you mentioned yesterday?

Reply:
The idea was to replace the existing `overrideProperty()` with something more generic. The idea is that we should use `wrapProperty()` every time when we augment an existing descriptor. `defineProperty()` should be only used when we want to define a new property (for example in webcompat scripts).

The new signature allows to use `wrapProperty()` in many more cases. For example, in cookie protection we're overriding `document.cookie` using `defineProperty()` because `overrideProperty()` only supports static values. On top of that, we should actually be overriding `Document.prototype.cookie` instead. And `wrapProperty()` would make sure that you're not changing the type of the descriptor as well.

`overrideProperty()` and `wrapProperty()` have very similar purpose, so it doesn't make sense to keep them both. The latter is more flexible, and the signature is closer to that of `defineProperty()`.

If you're worried about conflicts with the parallel branches, I can include these changes completely in https://github.com/duckduckgo/content-scope-scripts/pull/527, since you're the reviewer there anyways. Would this help?